aten/src/ATen/cpu/vec/vec_half.h

```diff
@@ -0,0 +1,59 @@
+#include <ATen/cpu/vec/intrinsics.h>
+
+#ifdef CPU_CAPABILITY_AVX512
+#if defined(__GNUC__)
+#define __at_align__ __attribute__((aligned(64)))
+#elif defined(_WIN32)
+#define __at_align__ __declspec(align(64))
+#else
+#define __at_align__
+#endif
+#else // CPU_CAPABILITY_AVX512
+#if defined(__GNUC__)
+#define __at_align__ __attribute__((aligned(32)))
+#elif defined(_WIN32)
+#define __at_align__ __declspec(align(32))
+#else
+#define __at_align__
+#endif
+#endif // CPU_CAPABILITY_AVX512
+
+namespace at {
+namespace vec {
+// See Note [CPU_CAPABILITY namespace]
+inline namespace CPU_CAPABILITY {
+
+#if defined(CPU_CAPABILITY_AVX2) || defined(CPU_CAPABILITY_AVX512)
+inline uint16_t float_to_half(float val) {
+#if defined(CPU_CAPABILITY_AVX2)
+  __m256 v = _mm256_set1_ps(val);
+  __m128i o =
+      _mm256_cvtps_ph(v, (_MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC));
+  return static_cast<std::uint16_t>(_mm_cvtsi128_si32(o));
```

Comment:
the lower 16 bits?

Reply:
Yes,  the lower 16 bits. Copied from https://github.com/pytorch/FBGEMM/blob/main/src/QuantUtilsAvx2.cc#L1556.